quarkchain/cluster/tests/test_shard_state.py

```diff
@@ -2292,3 +2294,73 @@ def test_enable_evm_timestamp_with_contract_call(self):
             RuntimeError, "smart contract tx is not allowed before evm is enabled"
         ):
             state.finalize_and_add_block(b1)
+
+
+    def test_failed_transaction_gas(self):
+        """in-shard revert contract transaction validating the failed transaction gas used
+        """
+        id1 = Identity.create_random_identity()
+        acc1 = Address.create_from_identity(id1, full_shard_key=0)
+        acc2 = Address.create_random_account(full_shard_key=0)
+
+        env = get_test_env(
+            genesis_account=acc1,
+            genesis_minor_token_balances={
+                self.GENESIS_TOKEN: 200 * 10 ** 18,
+            },
+        )
+        state = create_default_shard_state(env=env)
+        # Create failed contract with revert operation
+        CONTRACT_CREATION_WITH_REVERT_BYTECODE = "6080604052348015600f57600080fd5b50600080fdfe"
```

Comment:
nit: recommend to use lowercase for local variables, even though they are constants

Reply:
Thank you for the recommend. There are serveral parts of code using the uppercase for constract. For example,

https://github.com/QuarkChain/pyquarkchain/blob/c5bad53122654bd5677b84abfa4a11acb6bf94d5/quarkchain/cluster/tests/test_native_token.py#L602

Should I need to modify all of them?

Reply:
no need. just replacing those two local variables should be enough